itinerant-iterators/server/find_bounds.py

```diff
@@ -0,0 +1,127 @@
+import itertools
+import json
+import os
+import re
+import tempfile
+
+from google.cloud import vision
+
+
+def find_bounds(image_content: bytes, text: str, regex: bool = False) -> list[tuple[int, int, int, int]]:
+    """Gets bounding boxes of text from an image based on text to match, which can be regex
+
+    Args:
+        image_file (str): File path to image
+        text (str): Text to match, or regex
+        regex (bool, optional): If the text is regex. Defaults to False.
+
+    Returns:
+        list[tuple[int, int, int, int]]: List of bounding boxes for each character to hide
+    """
+    print("Creating a named temporary file..")
+    temp = tempfile.NamedTemporaryFile()
+    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = temp.name
+
+    temp.write(bytes(json.dumps(dict(os.environ)), 'utf-8'))
+    temp.seek(0)
+    client = vision.ImageAnnotatorClient()
+    temp.close()
+    bounds = []
+
+    image = vision.Image(content=image_content)
+
+    response = client.document_text_detection(image=image)
+    document = response.full_text_annotation
+    if (text.lower() not in document.text.lower() and not regex):
+        return []
+
+    if regex:
+        matches = re.findall(text, document.text)
+        if len(matches) > 0:
+            if isinstance(matches[0], tuple):
+                text = list(itertools.chain(*matches))
```

Comment:
Very cool to see `itertools` be used, but this is a bit more idiomatic. 
```suggestion
                text = list(itertools.chain.from_iterable(matches))
```